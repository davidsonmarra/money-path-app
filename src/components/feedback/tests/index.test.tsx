import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import Feedback, {
  FeedbackType,
  Props as FeedbackProps,
} from 'src/components/feedback';

const mockTitle = 'title';
const mockButtonText = 'buttonText';
const mockOnPress = jest.fn();
const renderComponent = ({
  title = mockTitle,
  buttonText = mockButtonText,
  onPress = mockOnPress,
  ...rest
}: Partial<FeedbackProps>) =>
  render(
    <Feedback
      title={title}
      buttonText={buttonText}
      onPress={onPress}
      {...rest}
    />,
  );

describe('Feedback', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders title', () => {
    renderComponent({});
    expect(screen.getByText(mockTitle)).toBeTruthy();
  });

  it('renders button text', () => {
    renderComponent({});
    expect(screen.getByText(mockButtonText)).toBeTruthy();
  });

  it('calls onPress when button is pressed', () => {
    renderComponent({});
    fireEvent.press(screen.getByText(mockButtonText));
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('renders description', () => {
    const mockDescription = 'description';
    renderComponent({description: mockDescription});
    expect(screen.getByText(mockDescription)).toBeTruthy();
  });

  it('renders success feedback', () => {
    renderComponent({type: FeedbackType.success});
    expect(screen.getByTestId('icon-check')).toBeTruthy();
  });

  it('renders error feedback', () => {
    renderComponent({type: FeedbackType.error});
    expect(screen.getByTestId('icon-error')).toBeTruthy();
  });

  it('renders warning feedback', () => {
    renderComponent({type: FeedbackType.warning});
    expect(screen.getByTestId('icon-warning')).toBeTruthy();
  });
});
