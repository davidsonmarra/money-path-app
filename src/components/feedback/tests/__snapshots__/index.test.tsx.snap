// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View />
  <View
    style={
      {
        "alignItems": "center",
        "gap": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 90,
            "height": 180,
            "justifyContent": "center",
            "marginBottom": 16,
            "width": 180,
          },
          {
            "backgroundColor": "#A6CD4E30",
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={100}
        bbWidth={100}
        fill="#A6CD4E"
        focusable={false}
        height={100}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 100,
              "width": 100,
            },
          ]
        }
        testID="icon-check"
        vbHeight={960}
        vbWidth={960}
        width={100}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4289121614,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m424-296 282-282-56-56-226 226-114-114-56 56zm56 216q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 24,
            "fontWeight": 600,
            "lineHeight": 28,
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      title
    </Text>
    <Text
      style={
        [
          [
            {
              "color": "#D9D9D9",
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontWeight": 400,
            },
          ],
          [
            {
              "textAlign": "center",
            },
            {
              "color": "#FFFFFFAA",
            },
          ],
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#EAC43D",
        "borderRadius": 12,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <View />
    <Text
      style={
        [
          [
            {
              "color": "#D9D9D9",
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "fontWeight": 600,
            },
          ],
          {
            "color": "#FFF",
          },
        ]
      }
    >
      buttonText
    </Text>
    <View />
  </View>
</View>
`;
